listenbrainz/webserver/static/js/src/CBReviewModal.tsx

```diff
@@ -0,0 +1,703 @@
+            <button
+              className="dropdown-toggle btn-transparent"
+              data-toggle="dropdown"
+              type="button"
+            >
+              {`${entityToReview.name} 
+              (${entityToReview.type.replace("_", " ")})`}
+              <span className="caret" />
+            </button>
+
+            <ul className="dropdown-menu" role="menu">
+              {/* Map entity to dropdown option button */}
+              {allEntities.map((entity) => {
+                if (entity) {
+                  return (
+                    <button
+                      key={entity.mbid}
+                      onClick={() => this.setState({ entityToReview: entity })}
+                      type="button"
+                    >
+                      {`${entity.name} (${entity.type.replace("_", " ")})`}
+                    </button>
+                  );
+                }
+                return <></>;
+              })}
+            </ul>
+          </span>
+          for <a href={this.CBBaseUrl}>CritiqueBrainz</a>.{this.CBInfoButton}
+        </div>
+
+        <div className="form-group">
+          <textarea
+            className="form-control"
+            id="review-text"
+            placeholder={`Review length must be at least ${this.minTextLength} characters.`}
+            value={textContent}
+            name="review-text"
+            onChange={this.handleTextInputChange}
+            rows={6}
+            style={{ resize: "vertical" }}
+            spellCheck="false"
+            required
+          />
+        </div>
+        <small style={{ display: "block", textAlign: "right" }}>
+          Words: {countWords(textContent)} / Chars: {textContent.length}
+        </small>
+
+        <div className="rating-container">
+          <b>Rating: </b>
+          <Rating
+            className="rating-stars"
+            onClick={(rate: number) => this.setState({ rating: rate })}
+            ratingValue={rating}
+            transition
+            size={20}
+            stars={5}
+          />
+        </div>
+
+        <div className="dropdown">
+          <b>Language: </b>
+          <select
+            id="language-selector"
+            value={language}
+            name="language"
+            onChange={this.handleInputChange}
+          >
+            {this.allLanguagesKeyValue.map((lang: any) => {
+              return (
+                <option key={lang[0]} value={lang[0]}>
+                  {lang[1]}
+                </option>
+              );
+            })}
+          </select>
+        </div>
+
+        <div className="checkbox">
+          <label>
+            <input
+              id="acceptLicense"
+              type="checkbox"
+              checked={acceptLicense}
+              name="acceptLicense"
+              onChange={this.handleInputChange}
+              required
+            />
+            <small>
+              &nbsp;You acknowledge and agree that your contributed reviews to
```

Comment:
This does not provide satisfactory UX. This needs to be fixed. If not right now, it should be dealt with later.

Reply:
I agree that the checkbox isn't ideal but I'm wondering what would be the best way to get around this. My understanding was that details of terms of service stuff should ideally be viewable upfront. Maybe it would be okay to move this into a popover, or give the text it's own page on CritiqueBrainz and just link it from the modal. @MonkeyDo what did you think?

Reply:
It's not a question I'm comfortable answering, to be honest. Maybe @alastair can give us his point of view?
(alastair, we're talking about the checkbox with the super long license text in the modal, see the video in the description)
Does the license text need to be in full as-is, or could it be a link to a page with the full text, or a popover, or something else?

However I don't think this preoccupation should block this PR.